### PR TITLE
wr-222: sockets integration

### PR DESCRIPTION
**Pull request by artemmatiushenko1**
🚀 This PR adds the following features:
- real-time display of comments on article view (+ notification)
- real-time display of reactions on article view/articles feed
- real-time display of new articles in the feed (+ notification)
image
image

### CODE REVIEW

File: frontend/src/libs/hooks/use-article-room/use-article-room.hook.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import { useSocketNamespace } from '../use-socket-namespace/use-socket-namespace.hook.js';
+
+const { NEW_COMMENT } = CommentsSocketEvent;
+const { NEW_REACTION } = ArticleReactionsSocketEvent;
+
+const useArticleRoom = (articleId: number): void => {
+  const dispatch = useAppDispatch();
+  const commentsSocketReference = useSocketNamespace(
+    SocketNamespace.COMMENTS,
+    SocketRoom.ARTICLE_$ID.replace(':id', articleId.toString()),
```
Comment (andanf-e) [inline frontend/src/libs/hooks/use-article-room/use-article-room.hook.ts]:
```suggestion
    configureString(SocketRoom.ARTICLE_$ID, { id: String(articleId) }),
```

File: frontend/src/libs/hooks/use-article-room/use-article-room.hook.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  );
+  const reactionsSocketReference = useSocketNamespace(
+    SocketNamespace.REACTIONS,
+    SocketRoom.ARTICLE_$ID.replace(':id', articleId.toString()),
```
Comment (andanf-e) [inline frontend/src/libs/hooks/use-article-room/use-article-room.hook.ts]:
```suggestion
    configureString(SocketRoom.ARTICLE_$ID, { id: String(articleId) }),
```

File: frontend/src/libs/hooks/use-articles-feed-room/use-articles-feed-room.hook.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
```
Comment (andanf-e) [inline frontend/src/libs/hooks/use-articles-feed-room/use-articles-feed-room.hook.ts]: Import them from already exported hooks folders

File: shared/src/libs/packages/socket/libs/enums/socket-event.enum.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+const SocketEvent = {
+  CONNECTION: 'connection',
+  JOIN_ROOM: 'join-room',
+  LEAVE_ROOM: 'leave-room',
+};
```
Comment (andanf-e) [inline shared/src/libs/packages/socket/libs/enums/socket-event.enum.ts]:
```suggestion
} as const;
```

**Review by andanf-e — DISMISSED** [review]

**Review by mhrechkin — DISMISSED** [review]

**Pull request merged**